extensions/sql-database-projects/src/models/project.ts

```diff
@@ -439,6 +440,18 @@ export class Project implements ISqlProject {
 
 		TelemetryReporter.sendActionEvent(TelemetryViews.ProjectController, TelemetryActions.updateProjectForRoundtrip);
 
+		// due to bug in DacFx.Projects, if a backup file already exists this will fail
+		// workaround is to rename the existing backup
+
+		if (await utils.exists(this.projectFilePath + '_backup')) {
+			let counter = 2;
+			while (await utils.exists(this.projectFilePath + '_backup' + counter)) {
+				counter++;
+			}
+
+			await fs.rename(this.projectFilePath + '_backup', this.projectFilePath + '_backup' + counter);
```

Comment:
Is it possible that the backup file has a Write hold, and the code won't be able to rename it?

Reply:
Yes, but the the calling code will display that error (once #22612 goes in).

This is just a work-around until the proper fix can be implemented in DacFx.Projects, but I think pushing a Projects update (and therefore STS) is more than we should do for this during Ask mode.